tests/FSharp.Compiler.UnitTests/PerfTests.fs

```diff
@@ -51,54 +47,54 @@ let ``Test request for parse and check doesn't check whole project`` () =
     printfn "ParseFile()..."
     let parseResults1 = checker.ParseFile(Project1.fileNames[5], Project1.fileSources2[5], Project1.parsingOptions)  |> Async.RunImmediate
     let pC, tC = FSharpChecker.ActualParseFileCount, FSharpChecker.ActualCheckFileCount
-    (pC - pB) |> shouldEqual 1
-    (tC - tB) |> shouldEqual 0
+    (pC - pB) |> Assert.shouldBe 1
+    (tC - tB) |> Assert.shouldBe 0
     printfn "checking backgroundParseCount.Value = %d" backgroundParseCount.Value
-    backgroundParseCount.Value |> shouldEqual 0
+    backgroundParseCount.Value |> Assert.shouldBe 0
     printfn "checking backgroundCheckCount.Value = %d" backgroundCheckCount.Value
-    backgroundCheckCount.Value |> shouldEqual 0
+    backgroundCheckCount.Value |> Assert.shouldBe 0
 
     printfn "CheckFileInProject()..."
     let checkResults1 = checker.CheckFileInProject(parseResults1, Project1.fileNames[5], 0, Project1.fileSources2[5], Project1.options)  |> Async.RunImmediate
     let pD, tD = FSharpChecker.ActualParseFileCount, FSharpChecker.ActualCheckFileCount
 
     printfn "checking background parsing happened...., backgroundParseCount.Value = %d" backgroundParseCount.Value
-    (backgroundParseCount.Value  >= 5) |> shouldEqual true // but note, the project does not get reparsed
+    (backgroundParseCount.Value  >= 5) |> Assert.shouldBeTrue // but note, the project does not get reparsed
     printfn "checking background typechecks happened...., backgroundCheckCount.Value = %d" backgroundCheckCount.Value
-    (backgroundCheckCount.Value  >= 5) |> shouldEqual true // only two extra typechecks of files
+    (backgroundCheckCount.Value  >= 5) |> Assert.shouldBeTrue // only two extra typechecks of files
 
     printfn "checking no extra background parsing...., backgroundParseCount.Value = %d" backgroundParseCount.Value
-    (backgroundParseCount.Value  <= 10) |> shouldEqual true // but note, the project does not get reparsed
+    (backgroundParseCount.Value  <= 10) |> Assert.shouldBeTrue // but note, the project does not get reparsed
     printfn "checking no extra background typechecks...., backgroundCheckCount.Value = %d" backgroundCheckCount.Value
-    (backgroundCheckCount.Value  <= 10) |> shouldEqual true // only two extra typechecks of files
+    (backgroundCheckCount.Value  <= 10) |> Assert.shouldBeTrue // only two extra typechecks of files
 
     printfn "checking (pD - pC) = %d" (pD - pC)
-    (pD - pC) |> shouldEqual 0
+    (pD - pC) |> Assert.shouldBe 0
     printfn "checking (tD - tC) = %d" (tD - tC)
-    (tD - tC) |> shouldEqual 1
+    (tD - tC) |> Assert.shouldBe 1
```

Comment:
I don't think there is any  value in changing all the trivial 'shouldEqual` occurrences altogether within these PRs, especially when the "previous" code was going via a helper function.

Just makes the diff and assessment of changes unnecessarily large, and easily leads to bugs.

(on example of a possible mistake: the existing 'shouldEqual' was written in  way to provide type safety and only compare elements of equal types, which is important especially with type inference. The Assert.shouldBe which was used instead allows arguments of different types, which can lead to inferring a more generic type (obj,IComparable,..) which then behaves differently.)

Firstly I do not think it matters that much when run, assertions should be possible to be combined from multiple libraries and it works.

If it does matter somewhere significantly (??), the function could be instead temporarily shadowed (and added via an 'open'), and then eventually the `shouldEqual` reimplemented to make it fit.

Reply:
Yes, this is a valid point, I didn't notice typed vs untyped comparison.

Well, we cannot keep FsUnit as it builds on top of NUnit which we are migrating from. So instead I added the typed comparison to the `Assert` file and opened it where needed hence minimizing the changes. I imagine that's what you meant by shadowing?

Reply:
Yep, that's good.

(also, I do not think we have to move away from FsUnit - usage of assertion library should be orthogonal choice to the usage of test runner. And, the inner workings of FsUnit.fs could be moved to use XUnit if that does make a difference)

Reply:
That's also fair, eventually we can stay with FsUnit, again it's just about consolidating all these helpers we have here and there.